plugins/module_utils/s3.py

```diff
@@ -151,3 +151,26 @@ def s3_extra_params(options, sigv4=False):
         config["signature_version"] = "s3v4"
     extra_params["config"] = config
     return extra_params
+
+
+# _list_backup_inventory_configurations is a workaround for a missing paginator for listing
+# bucket inventory configuration in boto3:
+# https://github.com/boto/botocore/blob/1.34.141/botocore/data/s3/2006-03-01/paginators-1.json
+def _list_bucket_inventory_configurations(client, bucket_name):
+    first_iteration = False
+    next_token = None
+
+    response = client.list_bucket_inventory_configurations(Bucket=bucket_name)
```

Comment:
```suggestion
    response = _list_bucket_inventory_configurations(Bucket=bucket_name)
```

It would be better to move this call out into a method of its own.  By doing so we can wrap the call in our "retry" decorator and better handle things like rate limits.

Generally what I'd do is rename your current method to "list_bucket_inventory_configurations" and then add a *new* method that just decorates the boto3 call:

```
AWSRetry.exponential_backoff(max_delay=120, catch_extra_error_codes=["NoSuchBucket", "OperationAborted"])
def _list_bucket_inventory_configurations(client, **params):
    return client.list_bucket_inventory_configurations(**params)
```